File: Data Structures & Algorithms/duplicate-integer/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let l = 0;
        let r = 1;
        while (r <= nums.length) {                   
            // console.log("l: " + l + " r: " + r);
            console.log(nums.length)
            if (nums[l] == nums[r]) {
                return true;
            }
            r++;    
            
            if (r == nums.length) {
                l++;
                r = l + 1;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # make a hashmap with the number and the count
        # add to res, k times
        
        # elements dictionary { element : count }
        eDict = {} 
        res = []

        for n in nums:
            if n not in eDict:
                eDict[n] = 1
            else:
                eDict[n] += 1
        
        # find greatest count, add to res 
        # then delete from hashMap
        while k > 0:
            maxFreq = max(eDict, key=eDict.get)
            res.append(maxFreq)
            del eDict[maxFreq]
            k -=1

        return res
            


